### PR TITLE
Add graphQL search support

### PR DESCRIPTION
**Pull request by shimish2**
1. Work on review changes.
2. Added more unit test case to increase code coverage.

### CODE REVIEW

File: pkg/api/ldap.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 	"fmt"
 
 	"github.com/anuvu/zot/errors"
+        // nolint (stylecheck)
```
Comment (tych0) [inline pkg/api/ldap.go:13]: Indentation looks off here, but other than that lgtm.

File: examples/config-test.json [hunk 2]
```diff
@@ -9,5 +9,8 @@
   },
   "log":{
     "level":"debug"
+  },
+  "extension":{
+    "dbPath":"./data/db/ZotSearch.db"
```
Comment (rchincha) [inline examples/config-test.json:14]: "extensions": {
"search": {
"dbPath": ...
}
}

File: pkg/api/config.go [hunk 3]
```diff
@@ -67,21 +69,37 @@ type LogConfig struct {
 	Output string
 }
 
+type ExtensionConfig struct {
+	Cve    *CveConfig
+	DBPath string
```
Comment (rchincha) [inline pkg/api/config.go:74]: Make this part of SearchConfig

File: examples/config-test.json (hunk 2 above)
Comment (shimish2) [reply examples/config-test.json:14]: Ok

File: pkg/api/config.go (hunk 3 above)
Comment (shimish2) [reply pkg/api/config.go:74]: Ok, I will change that

File: pkg/api/controller.go [hunk 4]
```diff
@@ -16,6 +16,10 @@ import (
 	"github.com/gorilla/mux"
 )
 
+const (
+	DBFilePath = "./data/db/ZotSearch.db"
```
Comment (rchincha) [inline pkg/api/controller.go:20]: Is this a separate directory to store the db? zot already has a cache.db it puts at top dir

File: pkg/api/routes.go [hunk 5]
```diff
@@ -102,8 +106,9 @@ func (rh *RouteHandler) SetupRoutes() {
 			rh.blobLockWrapper(rh.DeleteBlobUpload)).Methods("DELETE")
 		g.HandleFunc("/_catalog",
 			rh.ListRepositories).Methods("GET")
-		g.HandleFunc("/",
-			rh.CheckVersionSupport).Methods("GET")
+		g.HandleFunc("/", rh.CheckVersionSupport).Methods("GET")
```
Comment (rchincha) [inline pkg/api/routes.go:109]: don't change the indent here - it follows convention above

File: pkg/api/routes.go [hunk 6]
```diff
@@ -102,8 +106,9 @@ func (rh *RouteHandler) SetupRoutes() {
...
+		//nolint (lll)
+		g.HandleFunc("/query", gqlHandler.GraphQL(search.NewExecutableSchema(search.Config{Resolvers: &search.Resolver{DB: rh.c.Config.Extension.Cve.cvedb, Cve: &cveinfo.CveInfo{Log: rh.c.Log}}})))
```
Comment (rchincha) [inline pkg/api/routes.go:111]: break this into several lines? not possible? use nolint sparingly

Comment (rchincha) [inline pkg/api/routes.go:111]: also, is this URI path protected with authentication or anyone can access this route?

File: pkg/cli/root.go [hunk 7]
```diff
@@ -47,6 +47,7 @@ func NewRootCmd() *cobra.Command {
 					panic(errors.ErrBadConfig)
 				}
 			}
+			config.Extension.Cve = api.NewCveConfig(config.Extension.DBPath, config.Log)
```
Comment (rchincha) [inline pkg/cli/root.go:50]: Why not make this a part of NewConfig() itself?

File: pkg/extensions/search/utils/db.go [hunk 8]
```diff
@@ -0,0 +1,244 @@
+package cveinfo
```
Comment (rchincha) [inline pkg/extensions/search/utils/db.go:1]: this is a search/cve extension so call it cve.go - that it uses a db is a separate thing

File: pkg/api/controller.go (hunk 4 above)
Comment (shimish2) [reply pkg/api/controller.go:20]: Yes, I made that to store the data, if it has to be cache.db I can change that.

File: pkg/api/routes.go (hunk 5 above)
Comment (shimish2) [reply pkg/api/routes.go:109]: Ok, I will follow that

File: pkg/api/routes.go (hunk 6 above)
Comment (shimish2) [reply pkg/api/routes.go:111]: Ok, I will do that

File: pkg/api/controller.go (hunk 4 above)
Comment (rchincha) [reply pkg/api/controller.go:20]: No, call it search.db for now.

File: pkg/storage/storage.go [hunk 9]
```diff
@@ -152,7 +153,7 @@ func (is *ImageStore) ValidateRepo(name string) (bool, error) {
 		is.log.Error().Err(err).Str("dir", dir).Msg("unable to read directory")
 		return false, errors.ErrRepoNotFound
 	}
-
+	// nolint (gomnd)
```
Comment (rchincha) [inline pkg/storage/storage.go:156]: what is this nolint? you are not even touching this file right?

File: pkg/extensions/search/utils/db.go [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+}
+
+// nolint (interfacer)
+func Close(db *bbolt.DB) error {
+	err := db.Close()
+	return err
+}
+
+//CreateDB ...
+func (cve CveInfo) CreateDB(dbname string, db *bbolt.DB) bool {
```
Comment (rchincha) [inline pkg/extensions/search/utils/db.go:42]: Where is this being used in real code above?

File: pkg/api/routes.go (hunk 6 above)
Comment (shimish2) [reply pkg/api/routes.go:111]: I think it is authenticated since it is a part of same subrouter/route prefix.

Comment (rchincha) [reply pkg/api/routes.go:111]: Pls. cover it with a test case

File: pkg/storage/storage.go (hunk 9 above)
Comment (shimish2) [reply pkg/storage/storage.go:156]: Yes I am not touching this file but I got linting error during building, so I tried to fix that

File: pkg/extensions/search/utils/db.go (hunk 10 above)
Comment (shimish2) [reply pkg/extensions/search/utils/db.go:42]: Used in util.go InitSearch method

File: pkg/extensions/search/utils/db.go (hunk 8 above)
Comment (shimish2) [reply pkg/extensions/search/utils/db.go:1]: Ok, I will change that

File: pkg/api/routes.go (hunk 6 above)
Comment (shimish2) [reply pkg/api/routes.go:111]: Ok, I will do that

File: pkg/cli/root.go (hunk 7 above)
Comment (shimish2) [reply pkg/cli/root.go:50]: I thought our plan is to make CveSearch as a different service, its good to have different method for CveConfig and use wherever we need as of now.

**Pull request closed without merging**